src/IssueViz.UnitTests/Editor/LocationTagging/LocationTaggerTests.cs

```diff
@@ -0,0 +1,431 @@
+            var storeMock = new Mock<IIssueLocationStore>();
+            storeMock.SetupAdd(x => x.IssuesChanged += (sender, args) => { });
+
+            var bufferMock = CreateBufferMock(ValidBufferDocName);
+            bufferMock.SetupAdd(x => x.ChangedLowPriority += (sender, args) => { });
+
+            var testSubject = new LocationTagger(bufferMock.Object, storeMock.Object, ValidSpanCalculator, ValidLogger);
+
+            testSubject.Dispose();
+
+            storeMock.VerifyRemove(x => x.IssuesChanged -= It.IsAny<EventHandler<IssuesChangedEventArgs>>(), Times.Once);
+            bufferMock.VerifyRemove(x => x.ChangedLowPriority -= It.IsAny<EventHandler<TextContentChangedEventArgs>>(), Times.Once);
+        }
+
+        [TestMethod]
+        public void Ctor_SetsFilePath()
+        {
+            var testSubject = new LocationTagger(ValidBuffer, ValidStore, ValidSpanCalculator, ValidLogger);
+            testSubject.FilePath.Should().Be(ValidBufferDocName);
+        }
+
+        [TestMethod]
+        public void Ctor_InitialisesTags()
+        {
+            var existingSpan = CreateUniqueSpan();
+            var locVizWithSpan = CreateLocViz(existingSpan);
+            var locVizWithoutSpan = CreateLocViz(null);
+
+            var storeMock = new Mock<IIssueLocationStore>();
+            storeMock.Setup(x => x.GetLocations(ValidBufferDocName)).Returns(new[] { locVizWithSpan, locVizWithoutSpan });
+
+            var newSpan = CreateUniqueSpan();
+            var calculatorMock = new Mock<IIssueSpanCalculator>();
+            calculatorMock.Setup(x => x.CalculateSpan(It.IsAny<IAnalysisIssueLocation>(), It.IsAny<ITextSnapshot>())).Returns(newSpan);
+
+            var testSubject = new LocationTagger(ValidBuffer, storeMock.Object, calculatorMock.Object, ValidLogger);
+
+            testSubject.TagSpans.Should().NotBeNull();
+            testSubject.TagSpans.Count.Should().Be(2);
+            testSubject.TagSpans[0].Tag.Location.Should().BeSameAs(locVizWithSpan);
+            testSubject.TagSpans[1].Tag.Location.Should().BeSameAs(locVizWithoutSpan);
+
+            // Locations without spans should have one calculated for them
+            testSubject.TagSpans[0].Tag.Location.Span.Value.Should().Be(existingSpan);
+            testSubject.TagSpans[1].Tag.Location.Span.Value.Should().Be(newSpan);
+        }
+
+        [TestMethod]
+        public void OnIssuesChanged_BufferFileNotChanged_TagsChangedNotRaised()
+        {
+            var storeMock = new Mock<IIssueLocationStore>();
+            var testSubject = new LocationTagger(ValidBuffer, storeMock.Object, ValidSpanCalculator, ValidLogger);
+            var initialTags = testSubject.TagSpans;
+
+            SnapshotSpanEventArgs actualTagsChangedArgs = null;
+            testSubject.TagsChanged += (senders, args) => actualTagsChangedArgs = args;
+
+            RaiseIssuesChangedEvent(storeMock, "not the file being tracked.txt");
+
+            actualTagsChangedArgs.Should().BeNull();
```

Comment:
This is not the same as testing that the event was not raised, as it could've been raised with a null... why not create a mock and verify that it was not called?